tests/test_scripts.py

```diff
@@ -435,6 +435,72 @@ def test_filter_abund_6_trim_high_abund_Z():
     assert badseq in seqs       # should be there, untrimmed
 
 
+def test_filter_abund_7_retain_Ns():
+    # check that filter-abund retains sequences with Ns, and treats them as As.
+
+    infile = utils.get_temp_filename('test.fq')
+    in_dir = os.path.dirname(infile)
+
+    # copy test file over to test.fq & load into counting table
+    shutil.copyfile(utils.get_test_data('test-filter-abund-Ns.fq'), infile)
+    counting_ht = _make_counting(infile, K=17)
+
+    script = scriptpath('filter-abund.py')
+    args = ['-C', '3', counting_ht, infile]
+    utils.runscript(script, args, in_dir)
+
+    outfile = infile + '.abundfilt'
+    assert os.path.exists(outfile), outfile
+
+    # test for a sequence with an 'N' in it --
+    names = set([r.name for r in screed.open(outfile, parse_description=0)])
+    assert '895:1:37:17593:9954 1::FOO_withN' in names, names
+
+    # check to see if that 'N' was properly changed to an 'A'
+    seqs = set([r.sequence for r in screed.open(outfile)])
+    assert 'GGTTGACGGGGCTCAGGGGGCGGCTGACTCCGAG' not in seqs, seqs
+
+    # ...and that an 'N' remains in the output sequences
+    found_N = False
+    for s in seqs:
+        if 'N' in s:
+            found_N = True
```

Comment:
> Perhaps `found_N = 'N' in s`

I think this is more confusing to read than the 'if' statement, so -0 on
changing.